doc/man7/migration_guide.pod

```diff
@@ -0,0 +1,1791 @@
+=pod
+
+=head1 NAME
+
+migration_guide - OpenSSL migration guide
+
+=head1 SYNOPSIS
+
+See the individual manual pages for details.
+
+=head1 DESCRIPTION
+
+This guide details the changes required to migrate to new versions of OpenSSL.
+Currently this covers OpenSSL 3.0. For earlier versions refer to
+L<https://github.com/openssl/openssl/blob/master/CHANGES.md>
```

Comment:
How about a  reference here to the "crypto" man page for an overview of some of the key concepts in 3.0?